provider/azure/environ.go

```diff
@@ -1243,17 +1260,33 @@ func (env *azureEnviron) deleteVirtualMachine(
 
 	// The VM must be deleted first, to release the lock on its resources.
 	logger.Debugf("- deleting virtual machine (%s)", vmName)
-	vmResultCh, errCh := vmClient.Delete(env.resourceGroup, vmName, nil)
-	if result, err := <-vmResultCh, <-errCh; err != nil {
-		msg := "deleting virtual machine"
+	vmErrMsg := "deleting virtual machine"
+	vmFuture, err := vmClient.Delete(sdkCtx, env.resourceGroup, vmName)
+	if err != nil {
 		if errorutils.MaybeInvalidateCredential(err, ctx) {
-			return invalidatedCredential(err, msg)
+			return invalidatedCredential(err, vmErrMsg)
```

Comment:
This invalidatedCredential() wrapper func makes no sense - it just annotates the error with a message? Why do we need it instead of just calling errors.Annotate() directly.
Existing code, but....